### PR TITLE
[3pt] Pull Request: dev-enhance-alpha (no modification to hydroconditioning -- only evaluation techniques)

### PR DESCRIPTION
**Pull request by BradfordBates-NOAA**
This feature branch contains enhancements to tests/run_test_case.py, tests/utils/shared_functions.py and tests/aggregate_metrics.py to allow for 1) ahps test ids, 2) custom-provided inclusion masks, and 3) multiprocessed version of ble_autoeval (named synthesize_test_cases.py). This PR addresses issues #108, #50, and #95.

## Additions

- synthesize_test_cases.py

## Changes

- tests/run_test_case.py
- tests/utils/shared_functions.py
- tests/aggregate_metrics.py

## Testing

Check out branch and run:

`python foss_fim/tests/synthesize_test_cases.py -c DEV -v dev_c12582e_fr_copy -l /data/temp/parameter_adj_files/dev_fim_ble_forbrad.lst -d dev_c12582e_fr -o /data/temp/pull_requests/dev_enhance_alpha_pr_check/ -j 14`

In the ble_stats_comparison.csv, confirm that there is no difference in the metrics for all HUCs.

## Notes
Metrics were computed for dev_c12582e_fr using both the dev branch and dev-enhance-alpha. There was no observed change in metrics for any of the test hucs, meaning the branch's evaluation methodology is identical to the dev's methodology, and confirms that the new enhancements do not change the metrics. This is desired.

### CODE REVIEW

**Comment by TrevorGrout-NOAA** [comment]
I have reviewed this feature branch and approve pending one small modification.

I successfully ran the new evaluation command and was able to verify no changes in metrics occurred from current dev (dev_c12582e_fr).  The significant improvement to speeding up the evaluation calculations is nice! 

Additionally, although the AHPS evaluation is not exposed due to lack of benchmark data, Brad let me review some 'placeholder' data output and after reviewing the AHPS functionality I believe it is calculating correctly. Please make sure that 'moderate' and 'major' categories are added to the evaluation in addition to 'action' and 'minor'.  Once this is addressed I approve this PR.

**Review by TrevorGrout-NOAA — APPROVED** [review]

**Comment by BradfordBates-NOAA** [comment]
Moderate and Major categories were added back in. Thanks for approving!

**Pull request merged**
